File: MLR-spring/src/main/java/com/mlr/dto/LeakageContext.java
package com.mlr.dto;

import com.mlr.entity.Transaction;
import lombok.Data;
import java.util.List;
import java.util.Map;

@Data
public class LeakageContext {
    private String userId;
    private Double currentBalance;

  
    private List<Transaction> recentTransactions;
    private Double totalSpentLast30Days;
    private int debitCount;
    private Double avgDebit;


    private Double totalSpentPrevious30Days;


    private Double averageTransactionAmount;
    private int transactionCount;

    private Map<String, Double> categorySpending;
    private Double entertainmentSpent;

    private boolean hasDuplicateTransactions;
    private int salaryCreditCountLast30Days;
}

File: MLR/Rules/LeakageContext.cs
using MLR.DAO;

namespace MLR.Rules;

public class LeakageContext
{
    public string UserId { get; set; } = string.Empty;

    public decimal CurrentBalance { get; set; }

    public List<Transaction> RecentTransactions { get; set; } = new();

    public decimal TotalSpentLast30Days { get; set; }
    public decimal AverageTransactionAmount { get; set; }
    public int TransactionCount { get; set; }
    public Dictionary<string, decimal> CategorySpending { get; set; } = new();

    public bool HasDuplicateTransactions { get; set; }
    public decimal EntertainmentSpent { get; set; }
    public int DebitCount { get; set; }
    public decimal AvgDebit { get; set; }
    public int SalaryCreditCountLast30Days { get; set; }

    public decimal TotalSpentPrevious30Days { get; set; }
}
